### ISSUE TITLE
How I do it? I want solve buffering/lagging problems

### ISSUE THREAD

**Issue by Mgslt** [chrippa/livestreamer#1345]
http://docs.livestreamer.io/issues.html 

I want to solve the problem of buffering/lagging  but I do not know how to do... how should I do exactly ? Where do I put the commands ?

**Comment by StarWolf3000** [comment]
On livestreamer's command line, add:
`--player "/path/to/mplayer -cache <cachevalue>"` (in KB)
`--player "/path/to/mplayer2 -cache <cachevalue>"` (in KB)
`--player "/path/to/mpv --cache <cachevalue>"` (in KB)
`--player "/path/to/vlc --file-caching <cache-ms> --network-caching <cache-ms>"` (values in miliseconds)

**Comment by Mgslt** [comment]
Thanks... but I put just as you told me ?  with " "  at the beginning and end? In the file there is a lot of examples with # at the beginning , I put it down command at the end of them ?

**Comment by StarWolf3000** [comment]
Yes, you need them or single quotes ' , as this is needed to be treated as one single param for livestreamer. Without them, livestreamer would use them for itself.
Where'd you see the # ?

A complete command line can be as following:
`livestreamer whatyouwanttoplay --player "vlc --file-caching=5000 --network-caching=2000"`

**Comment by Mgslt** [comment]
I see  examples explaining how to use the commands...I already put what you gave me it in the end 
--player "/path/to/vlc --file-caching 10000 --network-caching 10000" is it correct? 

 Show console output from the video player
verbose-player

RTMP streams are downloaded using rtmpdump. Full path to the rtmpdump exe
 should be specified here.
rtmpdump=C:\Program Files\Livestreamer\rtmpdump\rtmpdump.exe

 Log level, default is info
loglevel=debug

 Number of threads to use when streaming HLS streams
hls-segment-threads=1

 Number of threads to use when streaming HDS streams
hds-segment-threads=1

--player "/path/to/vlc --file-caching 10000 --network-caching 10000"

**Comment by StarWolf3000** [comment]
I'm unsure about the =-signs for vlc's params, but yeah, this should work.

If vlc doesn't recognize the = then just replace them with a space (like you already did):
`--player "/path/to/vlc --file-caching 10000 --network-caching 10000"`

**Comment by Mgslt** [comment]
oh ok, thanks man :D

**Comment by ceberous** [comment]
> Playing -.
Reading from stdin...
Cache fill: 17.02% (1568672 bytes)   
libavformat version 56.40.101 (external)
TS file format detected.
Cache empty, consider increasing -cache and/or -cache-min. [performance issue]

different -cache-min and  -cache options for mplayer return the "cache empty" message no matter what. 

any other ideas for mplayer? 
it is constantly buffering no matter what. 

tried messing with each single "Stream transport options" as well. no difference really. just constant buffering.

**Comment by garoto** [comment]
For twitch I use this on my `livestreamerrc.twitch` file:
```
ringbuffer-size=196M
hls-live-edge=3
hls-segment-threads=4
```

No excessive buffering for me with these settings. Not sure if relevant, but I pipe to `mpv`.
